autobuild.sh

```diff
@@ -1,27 +1,11 @@
 # Checking for Operating System
 if [ "$(uname)" == "Darwin" ]; then
     # Install dependencies and Generate Makefile on Mac
-    chmod a+x scripts/prepare_env_mac.sh && scripts/prepare_env_mac.sh
+    bash scripts/build_mac.sh
```

Comment:
Why change this?

Reply:
The scripts used to only install software and set some environment variables, hence the name `prepare_env`. With the new build system, I've found it more convenient to run `configure` and `make` within each OS-specific shell script. This is mainly because `configure` has to be invoked differently on Mac OS to tell it to use the appropriate version of GCC.

Reply:
I suppose my question was more around using the bash command vs the chmod and then executing.